File: codigo-fonte/scala/manipulacao-de-listas/FiltrarNumerosPares.scala
object FiltrarNumerosPares extends App {

  val tempoInicial: Double = System.nanoTime

  val numeros = 1 to 1000000
  val resultado = numeros.view.filter(_ % 2 == 0)

  val tempoFinal: Double = System.nanoTime

  println(s"Resultado: ${resultado.size}")
  println(s"Tempo (ms): ${((tempoFinal - tempoInicial) / 1000000)}")

}

File: codigo-fonte/java/manipulacao-de-listas/FiltrarNumerosPares.java
import java.util.ArrayList;
import java.util.List;

public class FiltrarNumerosPares {

	public static void main(String[] args) {
		double tempoInicial = System.nanoTime();

		List<Long> resultado = new ArrayList<Long>();

		for (long numero = 1; numero <= 1000000; numero++) {
			if (numero % 2 == 0) {
				resultado.add(numero);
			}
		}

		double tempoFinal = System.nanoTime();

		System.out.println("Resultado: " + resultado.size());
		System.out.println("Tempo (ms): " + ((tempoFinal - tempoInicial) / 1000000));
	}

}

